City-of-Helsinki/parking-permits-admin-ui
# PV-402: Add search parameters in the orders view

## Description

This PR adds search parameters in the orders view as shown in [PV-402](https://helsinkisolutionoffice.atlassian.net/browse/PV-402) and applies some restyling as well.

New components:
- ZoneSelect, a dropdown menu for selecting the parking zone. Automatically fetches the parking zones from the API
- FilterOptions, which generates a list of checkboxes based on the given options and keeps track of the selected options
- OrdersSearch, includes all the search parameters and calls a callback function with the search parameters on submission

Changes:
- Add search parameters in orders view
- Store the search parameters in URL query parameters after performing a search
- Retrieve the search parameters from URL query parameters
- Make the order search lazy, i.e. when you open the view, it shows an empty state instead of automatically fetching the results
- Show an initial text in Table/DataTable when no data has been supplied yet (i.e. no query has been performed yet)
- Show a "no results" text in Table/DataTable when data has been supplied, but there are no items
- Show a loading spinner in Table/DataTable when loading
- Use Helsinki Grotesk font
- Restyle DataTable

## Context

The fetched orders need to be filtered by something. This feature adds parameters that allow filtering the search results.

[PV-402](https://helsinkisolutionoffice.atlassian.net/browse/PV-402)

Related PRs in the back-end:
[PV-402: Add support for searching orders #260](https://github.com/City-of-Helsinki/parking-permits/pull/260)
[PV-402: Return only distinct orders #266](https://github.com/City-of-Helsinki/parking-permits/pull/266)

## How Has This Been Tested?

Manually. No unit tests. Basically just clicking around while developing, very unorganized.

## Manual Testing Instructions for Reviewers

Fetch the branch, run the project locally, and hook it up with a back-end. I use my local back-end for this, but since the back-end PRs have been merged to develop branch, I imagine you could use the test back-end as well. If you use a local back-end, you'll need to add some orders there (if you haven't already).

Either way, here's stuff you can check:
- The page initially shows an empty table with "Syötä hakuehto" in it
- The page initially has no search parameters set
- Pressing the search button with no search parameters should return all orders (with no duplicates)
- Pressing the return key on the text search field should also perform a search
- Check that you can actually set the parameters and that the inputs themselves work (e.g. you can check a checkbox, enter text and so on)
- Check that the clear button appears after setting search parameters and that clicking the clear button removes the search parameters
- Perform some queries with search parameters, see that the parameters are correct... don't really have detailed instructions for this, but the filters themselves have been tested in the back-end, so you shouldn't need to comb through the results and see if they're correct
- After performing a search with parameters set, refresh the page and check that the search parameters are retained (aside from the text search fields, see "Known issues"), e.g. if low emission checkbox is checked -> it should still be checked after refreshing the page
- Check that the layout in general matches with the design in [PV-402](https://helsinkisolutionoffice.atlassian.net/browse/PV-402)

## Known issues

- Swedish translations are missing
- No validation for `DateInput` fields
- No error handling for invalid queries
- The `SearchInput` field cannot be set nor cleared; requested the HDS team to add support for this: [HDS-1417](https://helsinkisolutionoffice.atlassian.net/browse/HDS-1417)

## Screenshots

![image](https://user-images.githubusercontent.com/2333857/193783482-3e56a84b-49a0-4f30-9a66-a9225d299bb0.png)
Empty state

![image](https://user-images.githubusercontent.com/2333857/193783969-de4742a7-e297-4c80-92fa-405fba3cea81.png)
Results with some search parameters

![image](https://user-images.githubusercontent.com/2333857/193784216-55655a0f-b6a8-48b9-901f-3a7de96340e7.png)
No results found
